### ISSUE TITLE
Add algorithms in Ruby, C#, Golang and Rust

### ISSUE THREAD

**Issue by i-vishi** [VAR-solutions/Algorithms#1122]
Implement Data structures and algorithms in Ruby, Go, C# and Rust

**Comment by divyanshugit** [comment]
@i-vishi , Can I create a Pull request directly or do I need to first create a new issue for this?

**Comment by i-vishi** [comment]
Create a PR, no need to make another issue

**Comment by nits01** [comment]
I am trying to clone this repo after forking but its giving me errors.

**Comment by sahuaman83** [comment]
I have added Dijkstra's Algo in C# with documentation. Please Review!!

**Comment by Priyansh2001here** [comment]
Please merge #1232

**Comment by Priyansh2001here** [comment]
I can implement more but need some time

**Comment by Adarsh08X** [comment]
Hey can i add algorithm in c++ or python?

**Comment by i-vishi** [comment]
@Adarsh08X most of the algorithms are already implemented in c++ and python. If you want to add a new algorithm or want to improve the current code, you can go ahead

**Comment by Adarsh08X** [comment]
I've  added Kadane's algorithm in  [c#](https://github.com/VAR-solutions/Algorithms/pull/1348)

**Comment by varshaig1999** [comment]
Hey @i-vishi  can I make some changes to the current algorithms in c++ to make it better?

**Comment by i-vishi** [comment]
@varshaig1999 Yes, you can. If you find some errors in the current code or you want to add some explanation to the code you can make changes

**Comment by Mitali-laroia** [comment]
@i-vishi Added selection sort in c#.

**Comment by bcpeinhardt** [comment]
Hey there, 
Looking for a good first PR. Any particular algorithms need to be added in Rust?
